scripts/HPC_chatbot_preprocessor/chatbot_parser.py

```diff
@@ -0,0 +1,683 @@
+
+
+def replace_markdown_markers(curr_line, linklist, in_code_block):
+    """
+    function that replaces certain markdown structures with the equivalent used on the website
+
+    :param curr_line: the current line on which markdown structures need to be replaced
+    :param linklist: the list used to store links that need to be printed at the end of the file
+    :param in_code_block: boolean indicating whether the current line is part of a code block
+    :return curr_line: the adapted current line
+    :return linklist: the updated linklist
+    """
+
+    # replace images with an empty line
+    if re.search(r'(?i)!\[image]\(.*?\)', curr_line) or re.search(r'!\[]\(img/.*?.png\)', curr_line):
+        curr_line = ""
+
+    # replace links with a reference
+    matches = re.findall(r'\[(.*?)]\((.*?)\)', curr_line)
+    if matches:
+        for match in matches:
+            curr_line = curr_line.replace(f"[{match[0]}]({match[1]})", match[0] + "[" + str(len(linklist) + 1) + "]")
+            linklist.append(match[1])
+
+    # codeblock (with ``` -> always stands on a separate line, so line can be dropped)
+    if '```' in curr_line:
+        curr_line = ""
+
+    # structures within <>
+    match = re.findall(r'<(.*?)>', curr_line)
+    if match:
+        for i, content in enumerate(match):
+            syntax_words = ["pre", "b", "code", "sub", "br", "center", "p", "div", "u", "p", "i", "tt", "a", "t", "span"]  # make sure these are always lowercase
+            syntax_words_variations = list(chain.from_iterable([[element, element + "/", "/" + element] for element in syntax_words]))
+            syntax_words_style = [element + " style=.*" for element in syntax_words]
+
+            # add references for every link of format <a href=...>
+            if re.search(r'a href=.*', content):
+                link = content[8:-1]
+                curr_line = re.sub(f'<{content}>', "[" + str(len(linklist) + 1) + "]", curr_line)
+                linklist.append(link)
+
+            # drop the syntax words
+            elif content.lower() in syntax_words_variations:
+                curr_line = re.sub(f'<{content}>', "", curr_line)
+
+            # drop the version of the syntax_words followed by " style="
+            elif any(re.match(pattern, content) for pattern in syntax_words_style):
+                curr_line = re.sub(r'<.*?>', "", curr_line)
+
+            # drop markdown comments
+            elif re.fullmatch(r'!--.*?--', content):
+                curr_line = re.sub(r'<.*?>', "", curr_line)
+
+            # special case (ugly fix)
+            elif ' files</b' in content:
+                curr_line = re.sub(r'</b>', "", curr_line)
+
+            # keep the rest
+            else:
+                # print("<" + content + ">")
+                pass
+
+    # structures with !!! (info, tips, warnings)
+    if '!!!' in curr_line:
```

Comment:
`???` also exists for collapsible admonitions. I added these to the Linux tutorial in [my PR](https://github.com/hpcugent/vsc_user_docs/pull/647).